crates/flowistry_ide/src/mutations/mod.rs

```diff
@@ -0,0 +1,86 @@
+use anyhow::{Context, Result};
+use flowistry::{
+  mir::{aliases::Aliases, borrowck_facts::get_body_with_borrowck_facts, utils::SpanExt},
+  source_map::{self, location_to_spans, EnclosingHirSpans},
+};
+use log::debug;
+use rustc_hir::BodyId;
+use rustc_macros::Encodable;
+use rustc_middle::ty::TyCtxt;
+use rustc_span::Span;
+
+use self::find_mutations::find_mutations;
+use crate::{
+  analysis::{FlowistryAnalysis, FlowistryOutput, FlowistryResult},
+  range::{ranges_from_spans, Range},
+};
+
+mod find_mutations;
+
+struct MutationAnalysis {
+  range: Range,
+}
+
+#[derive(Debug, Clone, Encodable, Default)]
+pub struct MutationOutput {
+  pub ranges: Vec<Range>,
+  pub selected_spans: Vec<Range>,
+  pub body_span: Range,
+}
+
+impl FlowistryOutput for MutationOutput {
+  fn merge(&mut self, other: MutationOutput) {
+    self.ranges.extend(other.ranges);
+    self.body_span = other.body_span;
+    self.selected_spans.extend(other.selected_spans);
+  }
+}
+
+impl FlowistryAnalysis for MutationAnalysis {
+  type Output = MutationOutput;
+
+  fn locations(&self, tcx: TyCtxt) -> Result<Vec<Span>> {
+    Ok(vec![self.range.to_span(tcx.sess.source_map())?])
+  }
+
+  fn analyze_function(&mut self, tcx: TyCtxt, body_id: BodyId) -> Result<Self::Output> {
+    let def_id = tcx.hir().body_owner_def_id(body_id);
+    let body_with_facts = get_body_with_borrowck_facts(tcx, def_id);
+    let body = &body_with_facts.body;
+    let aliases = Aliases::build(tcx, def_id.to_def_id(), body_with_facts);
+
+    let source_map = tcx.sess.source_map();
+    let body_span = tcx.hir().body(body_id).value.span;
+    let targets =
+      source_map::span_to_places(tcx, body, body_span, self.range.to_span(source_map)?);
+
+    let (selected_place, _, _) = targets
```

Comment:
I changed `span_to_places` to return multiple places since that better captures the user's intent in some instances. I wouldn't just take the first one, instead just process all of them and union the result together.